pai-management/doc/how-to-write-pai-configuration.md

```diff
@@ -185,7 +185,7 @@ cluster:
     - ```docker-registry-domain```: E.g., gcr.io. If public，fill docker_registry_domain with the word "public".
     - ```docker-username```: The account of the docker registry
     - ```docker-password```: The password of the account
-    - ```docker-tag```: The image tag of the service. You could set the version here. Or just set latest here.
+    - ```docker-tag```: The image tag of the service, which should match the PAI release version, for example: `pai-0.6.y` for the release 0.6. Or use `latest` for the latest code.
```

Comment:
`pai-0.6.y` is the name of a temporary 'release candidate' branch that may eventually be deleted after the release (am I correct @sterowang ?) So I suggest we organize official images in this way:
- During the version release process: Create images with tag name **'pai-0.6.y'** (release candidate name) using the latest release candidate branch. Update these images whenever necessary. The `docker-tag` field in config file of the release candidate branch is set to be this tag name as well.
- Before exiting (announcing) the version release process: Create images with tag name **'pai-0.6.0'** (release name) using the code to be released (packaged to .tar.gz file). These images will never be updated. The `docker-tag` field in config file of the release package is set to this tag name as well.
- In master branch, the `docker-tag` field in config file is always set to **'latest'**. All images with tag 'latest' are built from the latest code in the master branch.

Reply:
My bad. I just realized RCs is not means to the intermediate states for images being verified, but some published versions for public testing. For example, when we want release a very big change version v0.6.0, maybe we can release v0.6.0-rc0, v0.6.0-rc1. Once we confidence enough for the last RC version, we publish the same version as gold version without RC surfix.

So for each image build, we can tag docker images with some build number based on the last commits to represent a specific git version. Once we publish the image, we can retag this build tag with release version. For example, we tag images this time with tag pai-0.6.0-12323123. Once we tagged code, we tag images with pai-0.6.0 and anounch it.

---
In reply to: [198025403](https://github.com/Microsoft/pai/pull/769#discussion_r198025403) [](ancestors = 198025403,198014564)